### PR TITLE
Smbv2: Possible bug fix. changes auth_protocol

### PR DESCRIPTION
**Pull request by Bargenish**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/36929

## Description
Change the auth_protocol to `ntlm` instead of the `negotiate` default type. (We do not support kerberos auth at the moment)
Seems that once the kerberos  auth falls back to ntlm, it uses the same session and the username argument is not being passed. (even though the ClientConfig was initialized with the username)

### Update
patch to an issue where the ClientConfig failed to be initiated when the dc argument was set and the negotiate auth flow fell to ntlm

### CODE REVIEW

File: Packs/SMB/Integrations/SMB_v2/SMB_v2.py [hunk 1]
```diff
@@ -47,7 +47,9 @@ def create_session(self, hostname: str = None, user: str = None, password: str =
             username=user or self._user,
             password=password or self._password,
             port=port or self._port,
-            encrypt=encrypt or self._encrypt)
+            encrypt=encrypt or self._encrypt,
+            auth_protocol='ntlm'
```
Comment (yaakovi) [inline Packs/SMB/Integrations/SMB_v2/SMB_v2.py:51]:
```suggestion
            auth_protocol='ntlm',
```

**Review by yaakovi — COMMENTED** [review]
ignoring the user name seems like a bug in the SDK, you can open an issue for them

File: Packs/SMB/Integrations/SMB_v2/SMB_v2.py [hunk 2]
```diff
@@ -216,6 +221,7 @@ def main():
         elif demisto.command() == 'smb-directory-remove':
             return_results(smb_rmdir(client, demisto.args()))
     except Exception as e:
+        traceback.print_exc()
```
Comment (yaakovi) [inline Packs/SMB/Integrations/SMB_v2/SMB_v2.py:224]:
```suggestion
        demisto.error(traceback.format_exc())
```

File: Packs/SMB/Integrations/SMB_v2/SMB_v2.py [hunk 3]
```diff
@@ -186,11 +188,14 @@ def main():
...
             client_guid = uuid.UUID(client_guid)
         except ValueError:
-            demisto.info(f'Failed to convert {client_guid} to a valid UUID string. Using a random generated UUID instead')
+            demisto.info(
+                f'Failed to convert {client_guid} to a valid UUID string. Using a random generated UUID instead')
             client_guid = None
 
-    smbclient.ClientConfig(username=user, password=password, require_secure_negotiate=verify, domain_controller=dc,
-                           client_guid=client_guid)
+    # Temporary workaround to an issue in the smbprotocol package.
```
Comment (yaakovi) [inline Packs/SMB/Integrations/SMB_v2/SMB_v2.py:195]: add a link to the github issue

File: Packs/SMB/ReleaseNotes/2_0_2.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### Server Message Block (SMB) v2
+- Maintenance and stability enhancements
```
Comment (yaakovi) [inline Packs/SMB/ReleaseNotes/2_0_2.md:4]:
```suggestion
- Fixed an issue where the negotiation in the authentication flow using the **Domain Controller** parameter failed. 
```

**Review by yaakovi — APPROVED** [review]

File: Packs/SMB/Integrations/SMB_v2/SMB_v2.py (hunk 3 above)
Comment (Bargenish) [reply Packs/SMB/Integrations/SMB_v2/SMB_v2.py:195]: It was closed due to the workaround. Should be resolved in the next release. I'll keep track of it.

**Pull request merged**